demos/app/iot_demo.c

```diff
@@ -118,19 +118,55 @@ int main( int argc,
     /* For a secured connection, set the members of the credentials. */
     if( demoArguments.securedConnection == true )
     {
-        /* Set credential paths. */
+        /* Set credential information. */
         credentials.pClientCert = demoArguments.pClientCertPath;
         credentials.pPrivateKey = demoArguments.pPrivateKeyPath;
         credentials.pRootCa = demoArguments.pRootCaPath;
+        
+        /* Set the MQTT username, as long as it's not empty or NULL. */
+        if( demoArguments.pUserName != NULL )
+        {
+            credentials.userNameSize = strlen( demoArguments.pUserName );
+            if( credentials.userNameSize > 0 )
+            {
+                credentials.pUserName = demoArguments.pUserName;
+            }
+            else
+            {
+                credentials.pUserName = NULL;
+            }
+        }
+
+        /* Set the MQTT password, as long as it's not empty or NULL. */
+        if( demoArguments.pPassword != NULL )
+        {
+            credentials.passwordSize = strlen( demoArguments.pPassword );
+            if( credentials.passwordSize > 0 )
```

Comment:
Do we need to enforce a maximum size for `credentials.passwordSize`?  The MQTT spec specifies that these [can't exceed 65535 bytes.](http://docs.oasis-open.org/mqtt/mqtt/v3.1.1/os/mqtt-v3.1.1-os.html#_Toc398718031)

Reply:
Good catch, thank you, fix is pending.

Reply:
Fixed.

Reply:
Consider also moving this (and the length check of user name above) to iot_demo_arguments.c too just so all checks are in one function.